### PR TITLE
implemented encode

### PR DESCRIPTION
**Pull request by krush11**
- added encode() in main()
- renamed files

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -3,13 +3,35 @@
...
+
+void encode(hufftree *root, string s = '\0')
+{
+    if(root == NULL)
+    {
+        freq2[find_key_value(root ->data)] = s;
+    }
+
+    encode(root ->right, s+'1');
+    encode(root ->left, s+'2');
```
Comment (krush11) [inline main.cpp:30]: `0` needs to be added for `root ->left`

Comment (Devika-j) [inline main.cpp:30]: Correction: s + '0' for left node

**Review by AKA2501 — COMMENTED** [review]
Correction: s + '0' for left node

**Comment by krush11** [comment]
@AKA2501 @Devika-j Please provide a review for PR closure

**Review by Devika-j — COMMENTED** [review]
Removed logical error

**Review by AKA2501 — APPROVED** [review]
Changes Reviewed

**Review by Devika-j — COMMENTED** [review]
changes reviewed

**Review by Devika-j — COMMENTED** [review]
encode function reviewed

**Review by Devika-j — APPROVED** [review]
changes reviewed

**Pull request merged**